Include parameterNotFoundAction in Validating Admission Policy documentation

Lahoussine-Bouhmou opened the pull request:
### **Description**

This pull request updates the Kubernetes documentation for Validating Admission Policies to include details about the `parameterNotFoundAction` field in `ValidatingAdmissionPolicyBinding`. 

#### Changes made:
- Added a detailed explanation of the `parameterNotFoundAction` field to the `paramRef` section in `validating-admission-policy.md`, describing its purpose, allowed values (`Allow` and `Deny`), and behavior.
- Updated YAML examples (`binding-with-param.yaml` and `binding-with-param-prod.yaml`) to include the `parameterNotFoundAction` field, ensuring consistency with the API requirements.
- Clarified the importance of setting `parameterNotFoundAction` for policy bindings that use parameter resources.
- Highlighted how the field interacts with scenarios where parameters are missing.

This change addresses the omission of `parameterNotFoundAction` from the documentation, which can lead to misconfigurations or confusion for users configuring admission policies.

dipesh-rawat commented:
@Lahoussine-Bouhmou Thanks you for your contribution.
Could you please sign the CLA before the PR can be reviewed.
You can follow the steps documented here: [CLA Steps](https://git.k8s.io/community/CLA.md)

dipesh-rawat reviewed (commented):
/easycla

_(To re-trigger EasyCLA bot check)_

dipesh-rawat commented on content/en/docs/reference/access-authn-authz/validating-admission-policy.md:
```diff
@@ -248,6 +248,49 @@ User is expected to have `read` access to the resources referenced by `paramKind
+
+One of `name` or `selector` must be set, but not both.
+
+**Note:** The `parameterNotFoundAction` field in `paramRef` is **required**. It specifies the action to take when no parameters are found matching the `paramRef`. If not specified, the policy binding may be considered invalid and will be ignored or could lead to unexpected behavior.
```
We could use `note` [shortcode](https://gohugo.io/content-management/shortcodes/) here. 
Refer the guidance on how to use it : https://kubernetes.io/docs/contribute/style/style-guide/#note

dipesh-rawat commented on content/en/docs/reference/access-authn-authz/validating-admission-policy.md:
```diff
@@ -248,6 +248,49 @@ User is expected to have `read` access to the resources referenced by `paramKind
+        environment: test
+    parameterNotFoundAction: Allow
+  validationActions:
+  - Deny
```
The ending triple backticks are missing here.

````suggestion
  - Deny
```  
````

dipesh-rawat commented:
/sig api-machinery

Lahoussine-Bouhmou commented:
@dipesh-rawat
Thank you for the feedback! I've did these commits as requested:
- Updated the note to use the shortcode.
- Fixed the missing triple backticks in the code block.

Let me know if there's anything else to adjust!

stevekuznetsov commented:
Fixes https://github.com/kubernetes/website/issues/48557

@dipesh-rawat or @jpbetz either of you have bandwidth for a review? The content here looks good to me.

stevekuznetsov commented:
@dipesh-rawat or @jpbetz either of you have bandwidth for a review? The content here looks good to me.

jpbetz reviewed (approved):
/approve
/lgtm

This looks fine technically. Thanks and apologies for the not noticing this till now.

tengqm commented:
/approve
